heron/api/src/java/com/twitter/heron/streamlet/impl/StreamletImpl.java

```diff
@@ -124,14 +125,21 @@ public String getName() {
     return name;
   }
 
-  protected void setDefaultNameIfNone(String prefix, Set<String> stageNames) {
+  /**
+   * Sets a default unique name to the Streamlet by type if it is not set.
+   * Otherwise, just checks its uniqueness.
+   * @param prefixType The name prefix type of this streamlet
+   * @param stageNames The collections of created streamlet/stage names
+   */
+  protected void setDefaultNameIfNone(StreamletNamePrefixType prefixType, Set<String> stageNames) {
     if (getName() == null) {
-      setName(defaultNameCalculator(prefix, stageNames));
+      setName(defaultNameCalculator(prefixType, stageNames));
     }
     if (stageNames.contains(getName())) {
       throw new RuntimeException(String.format(
           "The stage name %s is used multiple times in the same topology", getName()));
     }
+    stageNames.add(getName());
```

Comment:
Thanks @lucperkins for quick feedback. Sure, i will be removing enum changes form this PR due to addressed by other PR. 

In addition to test cases, this change (abstraction of `stageNames.add(getName()); `) can be part of this PR, if there is no concern ;)

Reply:
Yep, sounds good!